pkg/virt-api/rest/subresource.go

```diff
@@ -135,6 +136,44 @@ func (app *SubresourceAPIApp) putRequestHandler(request *restful.Request, respon
 	}
 }
 
+func (app *SubresourceAPIApp) httpGetRequestHandler(request *restful.Request, response *restful.Response, validate validation, getURL URLResolver, v interface{}) {
+	_, url, conn, err := app.prepareConnection(request, validate, getURL)
+	if err != nil {
+		log.Log.Errorf(prepConnectionErrFmt, err.Error())
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	resp, conErr := conn.Get(url, app.handlerTLSConfiguration)
+	if conErr != nil {
+		log.Log.Errorf(getRequestErrFmt, conErr.Error())
+		response.WriteError(http.StatusInternalServerError, conErr)
+		return
+	}
+
+	if err := json.Unmarshal([]byte(resp), &v); err != nil {
+		log.Log.Reason(err).Error("error unmarshalling response")
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	response.WriteEntity(v)
+}
+
+func (app *SubresourceAPIApp) decodeRequestBody(request *restful.Request, response *restful.Response, bodyStruct interface{}) bool {
```

Comment:
I like less the fact that this one returns a boolean.

Also, the is a subtle change, where you the body is passed originally as an address/pointer, and now by value.

Reply:
> Also, there is a subtle change, where the body is passed originally as an address/pointer, and now by value.

I think that should be fine. The `Decode` method also accepts ` interface{}` and the actual `bodyStruct` passed here is a pointer, e.g. ` bodyStruct := &v1.MigrateOptions{}`.

Reply:
Yea, it was a double pointer.